07.ls_object/lib/display.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require_relative 'file_status'
+
+class Display
+  COLUMN_COUNT = 3
+  SHORT_LIST_PADDING = 7
+  FILETYPES = {
+    'fifo' => 'p',
+    'characterSpecial' => 'c',
+    'directory' => 'd',
+    'blockSpecial' => 'b',
+    'file' => '_',
+    'link' => 'l',
+    'socket' => 's'
+  }.freeze
+  MODE_TABLE = {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }.freeze
+
+  def initialize(paths)
+    @file_statuses = paths.map { |path| FileStatus.new(path) }
+  end
+
+  def render(long: false)
+    if long
+      render_long_list
+    else
+      width = max_base_name_width + SHORT_LIST_PADDING
+      formated_file_statuses = @file_statuses.map { |file| file.base_name.ljust(width) }
+      file_count = @file_statuses.count
+      row_count = (file_count / COLUMN_COUNT.to_f).ceil
+      (COLUMN_COUNT - (file_count % COLUMN_COUNT)).times { formated_file_statuses << '' } if file_count % COLUMN_COUNT != 0
+      render_lines(formated_file_statuses, row_count)
+    end
+  end
+
+  def render_long_list
+    max_widths = find_max_widths
+    total = "total #{@file_statuses.sum(&:blocks)}"
+    long_list = @file_statuses.map { |status| organize(status, max_widths) }.map { |organized_status| format_row(organized_status) }
```

Comment:
```suggestion
    long_list = @file_statuses.map do |status|
      # 必要な整形をここで行う
    end
```

mapを2回繰り返すと、たとえばファイルが1万個ある場合、1万 x 2 = 2万回のループが回ります。
なので1回でまとめた方が実行効率が良いです。

Reply:
`map`2回くりかえしはループ処理が増えて危険信号だと認識しました。
効率性も意識していきたいです。